### PR TITLE
Add mise settings to preferences

### LINKED ISSUE #63: Add mise settings
Add mise settings panel 
image

Some options:
- load mise environment variables
- mise profile (important to know which mise files to load)
- Path to `mise` binary

### PR DESCRIPTION
**Pull request by hverlin**
Resolve #63

I am not sure what the exact behavior should be.
- do we use the global settings as default values when creating the run configuration
- or do we remove the settings from run configuration and always read the global settings?

### CODE REVIEW

**Comment by 134130** [comment]
> do we use the global settings as default values when creating the run configuration

Maybe we can disable the checkbox of run configuration's panel and add a comment "The configuration is overriden by Global Mise Settings"

**Comment by hverlin** [comment]
- using the global settings as default values when instantiating the run configuration
- Added a warning that the global configuration might be different from the run configuration
image

test

File: modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt [hunk 1]
```diff
@@ -1,46 +1,73 @@
...
+import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.options.SettingsEditor
 import com.jetbrains.nodejs.run.NodeJsRunConfiguration
 import org.jdom.Element
 
 class NodeRunConfigurationExtension : AbstractNodeRunConfigurationExtension() {
-    override fun getEditorTitle(): String = RunConfigurationSettingsEditor.EDITOR_TITLE
+    companion object {
+        private val LOG = Logger.getInstance(NodeRunConfigurationExtension::class.java)
+    }
```
Comment (134130) [inline modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt:19]: Is this logger works on your IDE?

I'm using Notification for debugging 😂

File: modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt [hunk 2]
```diff
@@ -1,46 +1,73 @@
...
+                    )
+                    configuration.envs = envs
+                }
 
-        val envs = config.envs.toMutableMap()
-        envs.putAll(MiseCmd.loadEnv(config.workingDirectory))
-        config.envs = envs
+                else -> {
+                    LOG.warn("Unsupported configuration type: ${configuration.javaClass}")
+                }
```
Comment (134130) [inline modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt:65]: 👍 I've found some issues that Linux-based IDEA's RunConfiguration implemented class is NpmRunConfiguration.

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -8,6 +8,13 @@
 
     <extensions defaultExtensionNs="com.intellij">
         <notificationGroup displayType="BALLOON" id="Mise"/>
+        <applicationService serviceImplementation="com.github.l34130.mise.settings.MiseSettings" />
+        <applicationConfigurable instance="com.github.l34130.mise.settings.MiseRunConfigurable" displayName="Mise" />
+        <applicationConfigurable
+                parentId="tools"
+                instance="com.github.l34130.mise.settings.MiseConfigurable"
+                id="com.github.l34130.mise.settings.MiseConfigurable"
+                displayName="Mise Settings"/>
```
Comment (134130) [inline src/main/resources/META-INF/plugin.xml:17]: FYI: I'm gonna make this configurable to project, not application.

**Review by 134130 — APPROVED** [review]
Thanks! Looks great to me!!

IMO, the run configuration is more detailed than global settings, global settings should be better not override run configuration's settings.
It will be annoying when we run gradle tasks, with several mise profiles, if global settings overrides it.

But never mind, I'll fix it later. 😃 

Additionally, it will be more nice if we can choose profile via combobox! (but I like simple textbox more 😆)

File: modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt (hunk 2 above)
Comment (hverlin) [reply modules/products/nodejs/src/main/kotlin/com/github/l34130/mise/runconfig/NodeRunConfigurationExtension.kt:65]: Same, for the npm one, I am not sure how it should work. I want to check how other env file plugins do it. Maybe it's possible to implement this generically for all languages

**Pull request merged**